### DISCUSSION TITLE
Container - how restart code server after changing config.yaml password setting ?

### DISCUSSION THREAD

**Discussion by stawen** [coder/code-server#5984] [Q&A]
Hi, i'm using you Docker Image `codercom/code-server:4.9.1`

I put in `config.yaml` a default password. But i say to my student to change this passord in this file.
But, the new password is not applied.

In your doc you write this : `sudo systemctl restart code-server@coder`
i have this : `sudo: systemctl: command not found`

So, how can i restart code-server ? (in docker context)

thank you

**Comment by lg-nineoneninethreenine** [comment]
Oh, that is a bit weird. When code-server is deployed on Docker, the password is taken as an environment variable afaik. Maybe try making a variable called PASSWORD and entering your student's password there?

**Reply by stawen** [reply]
Hi,

I wasn't specific enough. 

Yes i Know a random password is set when you start the container. But you must login in the terminal to know it. And i have a pipeline how deploy many container coder-server on Azure Private Container Instance (ACI). I deploy the docker for students. 
I don't want to connect to each container and then provide the password to each student
So in my dockerfile I put a custom config.yaml with the password `password`. The student login with this password `password` on their ACI I would like them to be able to change their password after login.

So they could change it in config.yaml file, but to be apply, coder-server must be restart.

So how restart it in a container (while keeping the start parameters) (And restart the container is not an Option, because with ACI i lost the IP address, so dns resolution will be broken)

**Reply by lg-nineoneninethreenine** [reply]
Well, I don't clearly understand what you said, I apologise, but is it like this? You have a container for each student, with a preset configuration. The student will know their password when they check it, but if they change it, they have to restart code-server. Is that what you meant? I hope I got your question right.

As far as I know(and have done, especially on a platform called Railway), to update code-server's version or variables, when it's a Docker container, like the password, you change it, of course, back up your files(you can use rsync, it does the job), restart the container, restore the files and you are good to go. The [documentation](https://coder.com/docs/code-server/latest) has this listed somewhere, I'm sure, so maybe that's all you need to do? I hope I helped you correctly. If you need more details/procedures on this, maybe @jsjoeio or @bpmct might know.

**Reply by stawen** [reply]
> Well, I don't clearly understand what you said, I apologise, but is it like this? You have a container for each student, with a preset configuration. The student will know their password when they check it, but if they change it, they have to restart code-server. Is that  what you meant? I hope I got your question right.

You understood. If we change the password in the config.yaml file, it is not use by code-server directly.
BUT, and it's very important, if we want the new password to be applied, code-server must be restart.
BUT (Again) **I don't want to restart the container**.  Only code-server in the container like 

```bash
sudo systemctl restart code-server@coder
```

or 

```bash
/usr/lib/code-server/lib/node /usr/lib/code-server restart
```

**Reply by lg-nineoneninethreenine** [reply]
But the caveat is, you can't do that. If you run code-server as an image, restarting the container is the way to go.

**Reply by lg-nineoneninethreenine** [reply]
But you could also try allocating VMs rather than containers, right?

**Reply by stawen** [reply]
Yes i can,  It's possible, but create a VM it's less light, we have a zero Vm politic in my company.

**Reply by lg-nineoneninethreenine** [reply]
Hmm... Hey, maybe try making a Ubuntu container, install code-server onto that and allow your students to use it? It's close to the real thing, innit?

**Reply by stanislas-draunet-maif** [reply]
Yeah maybe, I'll try.

**Reply by stanislas-draunet-maif** [reply]
So to do it, i made a script for changed password :   
- change password in config.yaml with sed
- a run backup file process
- apply this cli how force container to restart
```bash
kill -s HUP $(pgrep code-server)
```

And it's work.

**Reply by lg-nineoneninethreenine** [reply]
Excellent. It works as you expected?

**Reply by stanislas-draunet-maif** [reply]
yes perfectly
